### PR TITLE
chore(templates): add SSG template

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This adds an SSG-template using `vite-plugin-ssr`, I am however curious how we'd incorporate this in our current setup would we do

-> base/ssg?
--> base ? use-router? : none
--> useTS ?

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -36,6 +36,14 @@ const brandColor = /** @type {const} */ ([174, 128, 255]);
 						}
 					},
 				}),
+			type: () => prompts.select({
```
Comment (rschristian) [inline src/index.js:39]: I might go with `appType` or similar, just to be a bit more specific as to what it refers to

File: src/index.js [hunk 2]
```diff
@@ -132,6 +145,7 @@ async function scaffold(to, opts) {
 
 		const htmlPath = resolve(to, 'index.html');
 		const html = (await fs.readFile(htmlPath, 'utf-8')).replace('index.jsx', 'index.tsx');
+		// TODO: should we also rename all current jsx files to tsx?
```
Comment (rschristian) [inline src/index.js:148]: Yeah, I think I can move this to the template copy instead, and should, especially now that TS supports using `.ts` in import specifiers. Bad hack on my part, sorry.

**Review by rschristian — COMMENTED** [review]
Workflow looks great to me!

File: templates/ssr/package-lock.json
Comment (rschristian) [inline templates/ssr/package-lock.json:1]: Can delete this. Sorry, forgot to add nested lock files to the root `.gitignore`

File: src/index.js [hunk 3]
```diff
@@ -45,11 +53,11 @@ const brandColor = /** @type {const} */ ([174, 128, 255]);
 						{ value: 'ts', label: 'TypeScript' },
 					],
 				}),
-			useRouter: () =>
+			useRouter: ({ results }) => results.appType === 'spa' ?
 				prompts.confirm({
 					message: 'Use router?',
 					initialValue: false,
-				}),
+				}) : Promise.resolve(false),
```
Comment (JoviDeCroock) [inline src/index.js:60]: @rschristian do you know if there is a better way to do this? It looks like it goofs up our types

Comment (rschristian) [reply src/index.js:60]: I do not, no. [Clack does have some wonky types in places](https://github.com/natemoo-re/clack/issues/126) but it's been something I'm fine with `// @ts-ignore`'ing or whatever if the red squiggle is annoying for people.

We can ungroup the prompts, do something like [this](https://github.com/ascorbic/impala/blob/b2d1b5220527a66db98873ee38f17a73f9eea4de/packages/create-impala/src/cli.ts#L24-L35) to get better control flow, but it's perhaps a bit repetitive.

**Review by rschristian — APPROVED** [review]

**Pull request merged**